### ISSUE TITLE
Website inaccessible after login and Upgrade to 4.2.0

### ISSUE THREAD

**Issue by HeartBtz** [chevereto/chevereto#81]
Hello, 

I've just migrated to 4.2, and I can't access the platform after logging in. After the login form, which is functional, I get an empty page, with only the “Explore”, “Random” and “Research” tabs. 
It's no longer possible to take any action, and apart from clicking on Random, which returns an error, we're completely blocked. 

Best regards, 
HeartBtz

![Capture](https://github.com/user-attachments/assets/99d0ca46-8633-48ac-afdc-5bbe4d803a81)
![image](https://github.com/user-attachments/assets/ea88c249-5da6-40f9-ae7e-9151be6d2662)

**Comment by nikitakothari06** [comment]
Hi,

I am also facing the same issue on my test server with PHP 8.1 and Mysql 5.7 The installation went successful however on visiting "Admin Dashboard" it throws error. The script does not generate any logs as well.

![image](https://github.com/user-attachments/assets/86f8f967-14a1-452c-ab84-dabc8ff19bd5)

I also tried on Mysql 8.0 and the issue persists there as well. On visiting admin panel the login page loads but with the "Something went wrong" error.
![image](https://github.com/user-attachments/assets/c655005a-ae9d-4678-90fd-2e3ef6d2488d)

I tried enabling the logs in env.php file as well and it does not generate any log file.

**Comment by rodber** [comment]
Please follow the troubleshooting guide, I need to see the actual error.

https://v4-docs.chevereto.com/#troubleshooting

**Comment by HeartBtz** [comment]
After following troubleshooting, nothing in XRdebug, but executed
`sudo -u album php app/bin/legacy -C update`
gave me access to dashboard, but i lost all my website images (not albums, just images of the website)
![image](https://github.com/user-attachments/assets/2bd9a143-696a-43e6-b68e-553bb80fe655)
![image](https://github.com/user-attachments/assets/ae26ad8d-c896-4e9a-a991-e09cf2aa41d6)
![image](https://github.com/user-attachments/assets/62895a60-7293-447e-a812-92ddfd92744d)
![image](https://github.com/user-attachments/assets/82fa3eee-db4a-4ab3-9e92-23861d199c21)

Any ideas ? i cant upload new profile picture, cover etc... 

Thanks :)

**Comment by HeartBtz** [comment]
`VM124-MultiSite:/opt/Chevereto# find . -name logo_1719584898241_93de82.png
./content/images/system/logo_1719584898241_93de82.png`

**Comment by ne0ark** [comment]
I am also running into the same issue.

Here is the log from my docker:

```
# URI
/

# Method
HEAD

# Time
2024-10-25T16:15:05+00:00 [1729872905]

# Incident 671bc4093e118

# Backtrace
------------------------------------------------------------
0 /var/www/html/content/legacy/themes/Peafowl/header.php:401
{main}()
------------------------------------------------------------
1 /var/www/html/app/src/Legacy/G/functions-render.php:48
require(string(length=54))
------------------------------------------------------------
2 /var/www/html/app/src/Legacy/G/functions-render.php:54
Chevereto\Legacy\G\require_theme_file(string(length=6))
------------------------------------------------------------
3 /var/www/html/content/legacy/themes/Peafowl/views/404.php:11
Chevereto\Legacy\G\require_theme_header()
------------------------------------------------------------
4 /var/www/html/app/src/Legacy/G/Handler.php:523
require(string(length=57))
------------------------------------------------------------
5 /var/www/html/app/src/Legacy/G/Handler.php:188
Chevereto\Legacy\G\Handler->loadTemplate()
------------------------------------------------------------
6 /var/www/html/app/legacy/load/web.php:654
Chevereto\Legacy\G\Handler->__construct(bool(true), Closure(#15), Closure(#16))
------------------------------------------------------------
172.18.0.1 - - [25/Oct/2024:12:15:05 -0400] "HEAD / HTTP/1.1" 404 453 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/36.0.1985.67 Safari/537.36"
7 /var/www/html/app/legacy/entrypoints/index.php:43
require_once(string(length=37))
------------------------------------------------------------
8 /var/www/html/index.php:12
require_once(string(length=46))
------------------------------------------------------------

[671bc409584db] ErrorException thrown in /var/www/html/content/legacy/themes/Peafowl/header.php:401

# Message [Code #2]
Undefined array key "link_attr"
```

**Comment by rodber** [comment]
Thank you for your detailed reports; there’s a lot here to address.

---

### Regarding Login

The `4.2.0.zip` package is missing this file: [login_providers.php](https://github.com/chevereto/chevereto/blob/4.2/content/legacy/themes/Peafowl/snippets/quickty/login_providers.php), which is currently preventing `chevereto/chevereto` from functioning properly, including login access. 

To resolve this, add that file or use `touch content/legacy/themes/Peafowl/snippets/quickty/login_providers.php`—the system requires it, even if it’s empty.

---

### Regarding Tags

Some setups may be missing a database update, so ensure you’ve run the update both for files and the database. Try using the CLI with `app/bin/legacy -C update`.

If you’re using MySQL, note that I unintentionally built this for MariaDB, which caused compatibility issues I missed earlier. Fixing this will take some time, so I recommend waiting for `4.2.1` as I adjust this for both databases and remove the "latest" label from `4.2.0`.

---

### Regarding Assets

In Chevereto v4.2.0, asset paths can now be configured at **Dashboard > Settings > Assets**. Adjust the paths there, and I’ll work to keep these consistent in future releases to avoid any confusion.

---

Please keep testing and let me know if you come across any other issues—your feedback is invaluable!

**Comment by HeartBtz** [comment]
Hi there! 

Thanks for your reply! 
Indeed, modifying the asset directory allows you to retrieve these images. 

However, it seems that for some reason it didn't keep my .webp profile image. I can't find it anywhere... But anyway, I just got a 404 on this resource, I'll try to modify the profile picture, and it should be fine.

If I find any other errors, I'll let you know, thanks!

**Comment by HeartBtz** [comment]
update, after trying to re-download a profile picture, it also seems to come up with a 404. 
![image](https://github.com/user-attachments/assets/63792345-0f2b-4144-bd3b-9cda51a3cf84)

I've set my asset path to /opt/Chevereto/content/images/system , and it seems to recreate a sort of tree structure from there: 
```shell
VM124-MultiSite:/opt/Chevereto/content/images/system# ll
total 36948
-rwxrwxrwx 1 album album 31862427  3 sept. 20:10 consent-screen_cover_1725387026020_8da015.png
drwxrwxrwx 3 album album     4096 25 oct.  23:59 content
drwxrwxrwx 2 album album     4096 25 oct.  08:20 default
-rwxrwxrwx 1 album album   140095 28 juin  16:26 favicon_1719584812756_f197d5.png
-rwxrwxrwx 1 album album  5671060 28 juin  16:21 home_cover_1719584497268_73210d.png
-rwxrwxrwx 1 album album    86866 28 juin  16:28 logo_1719584898241_93de82.png
-rwxrwxrwx 1 album album    57301 28 juin  16:32 watermark_1719585129746_7f5261.png
VM124-MultiSite:/opt/Chevereto/content/images/system# cd content/
VM124-MultiSite:/opt/Chevereto/content/images/system/content/images/users/FsS# namei -ol /opt/Chevereto/content/images/system/content/images/users/FsS/av_1729886435.png
f: /opt/Chevereto/content/images/system/content/images/users/FsS/av_1729886435.png
drwxr-xr-x root     root     /
drwxrwxrwx www-data www-data opt
drwxr-x--- album    album    Chevereto
drwxrwxrwx album    album    content
drwxrwxrwx album    album    images
drwxrwxrwx album    album    system
drwxrwxrwx album    album    content
drwxrwxrwx album    album    images
drwxrwxrwx album    album    users
drwxrwxrwx album    album    FsS
-rw-r--r-- album    album    av_1729886435.png
VM124-MultiSite:/opt/Chevereto/content/images/system/content/images/users/FsS#
```

**Comment by ne0ark** [comment]
> app/bin/legacy -C update

Docker is still giving same error:

```
# URI
/

# Method
GET

# Time
2024-10-25T22:33:18+00:00 [1729895598]

# Incident 671c1cae2bf13

# Backtrace
------------------------------------------------------------
0 /var/www/html/content/legacy/themes/Peafowl/header.php:401
{main}()
------------------------------------------------------------
1 /var/www/html/app/src/Legacy/G/functions-render.php:48
require(string(length=54))
------------------------------------------------------------
2 /var/www/html/app/src/Legacy/G/functions-render.php:54
Chevereto\Legacy\G\require_theme_file(string(length=6))
------------------------------------------------------------
3 /var/www/html/content/legacy/themes/Peafowl/views/404.php:11
Chevereto\Legacy\G\require_theme_header()
172.18.0.1 - - [25/Oct/2024:18:33:18 -0400] "GET / HTTP/1.1" 404 24988 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/36.0.1985.67 Safari/537.36"
------------------------------------------------------------
4 /var/www/html/app/src/Legacy/G/Handler.php:523
require(string(length=57))
------------------------------------------------------------
5 /var/www/html/app/src/Legacy/G/Handler.php:188
Chevereto\Legacy\G\Handler->loadTemplate()
------------------------------------------------------------
6 /var/www/html/app/legacy/load/web.php:654
Chevereto\Legacy\G\Handler->__construct(bool(true), Closure(#15), Closure(#16))
------------------------------------------------------------
7 /var/www/html/app/legacy/entrypoints/index.php:43
require_once(string(length=37))
------------------------------------------------------------
8 /var/www/html/index.php:12
require_once(string(length=46))
------------------------------------------------------------

[671c1caeee915] ErrorException thrown in /var/www/html/content/legacy/themes/Peafowl/header.php:401

# Message [Code #2]
Undefined array key "link_attr"

# URI
/

# Method
HEAD

# Time
2024-10-25T22:33:18+00:00 [1729895598]

# Incident 671c1caeee915

# Backtrace
------------------------------------------------------------
0 /var/www/html/content/legacy/themes/Peafowl/header.php:401
{main}()
------------------------------------------------------------
1 /var/www/html/app/src/Legacy/G/functions-render.php:48
require(string(length=54))
------------------------------------------------------------
2 /var/www/html/app/src/Legacy/G/functions-render.php:54
Chevereto\Legacy\G\require_theme_file(string(length=6))
------------------------------------------------------------
3 /var/www/html/content/legacy/themes/Peafowl/views/404.php:11
Chevereto\Legacy\G\require_theme_header()
------------------------------------------------------------
4 /var/www/html/app/src/Legacy/G/Handler.php:523
require(string(length=57))
------------------------------------------------------------
5 /var/www/html/app/src/Legacy/G/Handler.php:188
Chevereto\Legacy\G\Handler->loadTemplate()
------------------------------------------------------------
172.18.0.1 - - [25/Oct/2024:18:33:18 -0400] "HEAD / HTTP/1.1" 404 453 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/36.0.1985.67 Safari/537.36"
6 /var/www/html/app/legacy/load/web.php:654
Chevereto\Legacy\G\Handler->__construct(bool(true), Closure(#15), Closure(#16))
------------------------------------------------------------
7 /var/www/html/app/legacy/entrypoints/index.php:43
require_once(string(length=37))
------------------------------------------------------------
8 /var/www/html/index.php:12
require_once(string(length=46))
------------------------------------------------------------

[671c1caf16d9e] ErrorException thrown in /var/www/html/content/legacy/themes/Peafowl/header.php:401

# Message [Code #2]
Undefined array key "link_attr"

# URI
/

# Method
GET

# Time
2024-10-25T22:33:19+00:00 [1729895599]

# Incident 671c1caf16d9e

# Backtrace
------------------------------------------------------------
0 /var/www/html/content/legacy/themes/Peafowl/header.php:401
{main}()
------------------------------------------------------------
1 /var/www/html/app/src/Legacy/G/functions-render.php:48
require(string(length=54))
------------------------------------------------------------
2 /var/www/html/app/src/Legacy/G/functions-render.php:54
Chevereto\Legacy\G\require_theme_file(string(length=6))
------------------------------------------------------------
3 /var/www/html/content/legacy/themes/Peafowl/views/404.php:11
Chevereto\Legacy\G\require_theme_header()
------------------------------------------------------------
4 /var/www/html/app/src/Legacy/G/Handler.php:523
require(string(length=57))
------------------------------------------------------------
5 /var/www/html/app/src/Legacy/G/Handler.php:188
Chevereto\Legacy\G\Handler->loadTemplate()
------------------------------------------------------------
6 /var/www/html/app/legacy/load/web.php:654
Chevereto\Legacy\G\Handler->__construct(bool(true), Closure(#15), Closure(#16))
------------------------------------------------------------
7 /var/www/html/app/legacy/entrypoints/index.php:43
require_once(string(length=37))
------------------------------------------------------------
8 /var/www/html/index.php:12
require_once(string(length=46))
------------------------------------------------------------`
``

**Comment by rodber** [comment]
Thank you for your continued reports! I've committed a few patches that should address the issues you've been encountering:

---

### MySQL Compatibility
- Commit [f9071bc](https://github.com/chevereto/chevereto/commit/f9071bce3d7090accf9439fc0c0a0f045d494093) adds support for tags functionality on MySQL. Please give it a try and let me know how it goes.

### Unresponsive Pages (White/Purple Screens)
- Commits [53449a3](https://github.com/chevereto/chevereto/commit/53449a3f57663f55723de8d3e194d57a7edb9a13) and [651c863](https://github.com/chevereto/chevereto/commit/651c863cc2ed724c454a464ce10717990b27d92f) address issues with missing or outdated files causing sites to become unresponsive.

### Asset Issues
- Head over to **Dashboard > Settings > Asset Storage** and adjust the config by removing `images/_assets/` from the path. That should resolve any asset loading issues.

--- 

Thank you again for your feedback, and let me know if further issues arise!

**Comment by ne0ark** [comment]
> Thank you for your continued reports! I've committed a few patches that should address the issues you've been encountering:
> 
> ### MySQL Compatibility
> * Commit [f9071bc](https://github.com/chevereto/chevereto/commit/f9071bce3d7090accf9439fc0c0a0f045d494093) adds support for tags functionality on MySQL. Please give it a try and let me know how it goes.
> 
> ### Unresponsive Pages (White/Purple Screens)
> * Commits [53449a3](https://github.com/chevereto/chevereto/commit/53449a3f57663f55723de8d3e194d57a7edb9a13) and [651c863](https://github.com/chevereto/chevereto/commit/651c863cc2ed724c454a464ce10717990b27d92f) address issues with missing or outdated files causing sites to become unresponsive.
> 
> ### Asset Issues
> * Head over to **Dashboard > Settings > Asset Storage** and adjust the config by removing `images/_assets/` from the path. That should resolve any asset loading issues.
> 
> Thank you again for your feedback, and let me know if further issues arise!

Is there a docker version of Dev I can test?

**Comment by rodber** [comment]
@ne0ark To test source changes on Docker, clone the `chevereto/docker` repository. Inside the `chevereto` folder, clone the `chevereto/chevereto` repository. When building the image, use `make image-custom`—this tells the build script to use the contents of `./chevereto` as the application filesystem.

**Comment by ne0ark** [comment]
> @ne0ark To test source changes on Docker, clone the `chevereto/docker` repository. Inside the `chevereto` folder, clone the `chevereto/chevereto` repository. When building the image, use `make image-custom`—this tells the build script to use the contents of `./chevereto` as the application filesystem.

Hello, I am using unraid its not easy to build custom docker containers. Is there a way you can push out Dev or testing branch for Docker so it is easy to pull from?

**Comment by rodber** [comment]
@ne0ark Simply replace the updated app filesystem on the container directly.

**Comment by rodber** [comment]
@ne0ark The issue you are experiencing is not exclusive to Unraid, so there’s no need to worry about testing it there. The problem has been addressed and will be fixed in the upcoming 4.2.1 release.

**Comment by rodber** [comment]
Chevereto v4.2.1 has been announced! You can follow all updates and details in [this topic](https://chevereto.com/community/threads/chevereto-v4-2-1-announcement.15675/).

**Comment by rodber** [comment]
@ne0ark @HeartBtz @nikitakothari06 A new release is now available, and it should address all the issues you've encountered. Please give it a try, and feel free to let me know if everything is working smoothly or if there's anything else you need help with!

https://github.com/chevereto/chevereto/releases/tag/4.2.1

**Comment by JasonHK** [comment]
> @ne0ark @HeartBtz @nikitakothari06 A new release is now available, and it should address all the issues you've encountered. Please give it a try, and feel free to let me know if everything is working smoothly or if there's anything else you need help with!
> 
> https://github.com/chevereto/chevereto/releases/tag/4.2.1

I can confirm 4.2.1 fixed the problem

**Comment by nikitakothari06** [comment]
Hi,

Still getting the same error on installing 4.2.1 version as well. I am trying the installation on PHP 8.1, Mysql 5.7
Not able to see the admin panel post filling all the details.

![image](https://github.com/user-attachments/assets/574cb9f5-79eb-4942-b3c8-72d089b713c8)


Attaching the log file here.
[chevereto.log](https://github.com/user-attachments/files/17633356/chevereto.log)

**Comment by rodber** [comment]
@nikitakothari06 The ENV variables must be of type **string**, somewhere you are passing an integer.

Will consider to auto convert everything to string on load, but at this time the system is strict.

**Comment by nikitakothari06** [comment]
Hi,

Thank you.

I am getting following error on Mysql 5.7
PDOException thrown in [[path]]/app/src/Legacy/G/DB.php:163

# Message [Code #42000]
SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'VISIBLE, KEY `tag_user_count` (`tag_user_count`), KEY `tag_user_last_used_da' at line 8

Then I checked the requirement of mysql and it has changed to 8.0+ I tried on another server with mysql 8.0 and its working fine now.

**Comment by rodber** [comment]
Thank you for the heads-up, @nikitakothari06.

Ideally, it should still function with MySQL 5, as I haven’t officially dropped support yet (it runs, but some nested album features are limited). I’ve included a deprecation message that appears as a system notification.

I’ll add a MySQL 5 test to prevent any unnecessary error screens and ensure it defaults to the system notice.

**Comment by rodber** [comment]
@nikitakothari06 @JasonHK Another release is now available ([4.2.2](https://github.com/chevereto/chevereto/releases/tag/4.2.2)) it should address plenty issues.

**Comment by nikitakothari06** [comment]
Hi,

I did try the 4.2.2 version but I am unable to start the installer.

Throwing "Something went Wrong" error.
![image](https://github.com/user-attachments/assets/1412c147-af56-4517-94f3-46d0dd3ecc31)

Trying on my test server with PHP 8.1 and mysql 8.0

Tried renaming env-default.php to env.php and setting following values to check the error but it's not throwing any error.
'CHEVERETO_DEBUG_LEVEL' => '3',
'CHEVERETO_ENABLE_DEBUG' => '1',

**Comment by rodber** [comment]
@nikitakothari06 If "Debug with logging" didn't yield results, the best approach is to try a different debugging method. Chevereto offers multiple options here: [Debug options](https://v4-docs.chevereto.com/developer/how-to/debug.html).

For your specific situation, I'd recommend following the [Debug (development)](https://v4-docs.chevereto.com/developer/how-to/debug.html#debug-development) guide, as this method should work even at very early stages of bootstrapping. This might help you pinpoint the issue more effectively.

**Comment by nikitakothari06** [comment]
Hi,

I tried installing on my other server with PHP 8.2 and Mysql 8.0, the installation is going well.  

However on populating database the script loads for long and at the end shows "The page does not exist" error but the installation completes successfully.

![image](https://github.com/user-attachments/assets/44f16466-572a-4415-a19e-f86ffdcb14a6)

What is causing this ?

**Comment by rodber** [comment]
@nikitakothari06, with the information provided, it's impossible to determine the cause as I can't replicate the context to trigger that error. Unfortunately, we don't have other users reporting this issue, so we can't identify a common factor to investigate.

Until we get more cases, the only way to fix this is by inspecting the server directly.

**Comment by rodber** [comment]
I've just published a new release (4.2.3) which should fix some of the issues mentioned here. Unfortunately, managing multiple issues in a single thread is challenging, so I must move this to discussions. If you encounter any further issues, please don't hesitate to create a new issue.

Many thanks for using Chevereto.
